doc/specs/wiki.md

```diff
@@ -0,0 +1,391 @@
+3. The user fills out the fields and user class read/edit restrictions, and hits submit.
+
+**Non-Functional Requirements**
+
+* BBCode must be stored in the database and rendered at page load
+* Each section must have a [+] button to add another section.
+* Each section must have a [++] button to add a sub-section.
+
+### 2.0 Viewing an article
+
+**Primary Actor:** A user
+
+**Brief:** Wiki articles must be able to be viewed.
+
+**Preconditions:** The user has the appropriate user class to read the article, or has the wiki management permission enabled.
+
+**Trigger:** The user navigates to the article's URL.
+
+**Postcondition:**
+
+* The article is rendered.
+
+**Basic Flow:**
+
+1. The user navigates to the URL of the article.
+2. The article is rendered.
+
+**Non-Functional Requirements:**
+
+* The table of contents must be auto-generated from the sections and sub-sections.
+* The tags, aliases, and dedicated words must be presented in a user-friendly manner.
+
+### 3.0 Editing an article
+
+**Primary Actor:** A user
+
+**Brief:** Editing articles allows mistakes and misinformation to be corrected, and current information to be expanded upon.
+
+**Preconditions:** The user has the appropriate user class to edit the article, or has the wiki management permission enabled.
+
+**Trigger:** The user clicks the edit link on the article's page.
+
+**Postcondition:**
+
+* The edits to the article are saved.
+
+**Basic Flow:**
+
+1. The user clicks the edit link on the article's page.
+2. The user clicks submit.
+3. The changes are saved to the database, and the user is redirected to the updated article.
+
+**Functional Requirements:**
+
+* The article's revision history must be updated with the changes.
+
+**Non-Functional Requirements:**
+
+* The article editing template must be the same as the article creation template, except with all of the fields filled in.
+
+### 4.0 Adding aliases
+
+**Primary Actor:** A user
+
+**Brief:** Alises allow wiki articles to be easily searched. Managing aliases ensures they are accurate and thorough.
+
+**Preconditions:**
+
+* The user is currently viewing the article.
+* The user has the appropriate user class to edit the article.
+
+**Trigger:** The user clicks the submit button next to the alias textbox.
+
+**Postcondition:**
+
+* The alias is added to the article and committed to the database.
+
+**Basic Flow:**
+
+1. The user enters the new alias into the textbox.
+2. The user hits submit, and the alias is added to the article.
+
+**Applicable business rules:**
+
+### When the user hits submit, the system must ensure that the alias isn't a duplicate, and that it is not empty.
+
+### 4.1 Removing aliases
+
+**Primary Actor:** A user
+
+**Brief:** Aliases must be able to ensure relevancy and accuracy.
+
+**Preconditions:**
+
+* The user is currently viewing the article.
+* The user has the appropriate user class to edit the article.
+
+**Trigger:** The user clicks the X button next to an alias.
```

Comment:
Lol, nice reference. Not sure what you mean by it but I removed all of the specifics related to UI elements.

Resolved.